### PR TITLE
Mask override param in api calls

### PR DESCRIPTION
**Pull request by petchells**
Avoid printing sensitive information to the Jetty logs. Reuse PasswordFilter which is already used in audit/qry/upd logs.

### CODE REVIEW

File: whois-api/src/test/java/net/ripe/db/whois/api/log/JettyRequestLogTestIntegration.java [hunk 1]
```diff
@@ -101,19 +103,43 @@ public void password_filtered() throws Exception {
...
 
 
-        String actual = fileToString(getRequestLogFilename());
+        String actual = fileToString(getRequestLog());
         assertThat(actual, containsString("GET /whois/test/person/TP1-TEST?password=FILTERED"));
         assertThat(actual, not(containsString("some-api_key-123")));
     }
 
+    @Test
+    public void password_and_override_filtered() throws Exception {
```
Comment (eshryane) [inline whois-api/src/test/java/net/ripe/db/whois/api/log/JettyRequestLogTestIntegration.java:112]: Add tests for mixed-case (and/or uppercase) password and override. Query parameters are case-sensitive, but if a user mixed the case by mistake we don't want sensitive data in the log.

File: whois-api/src/test/java/net/ripe/db/whois/api/log/JettyRequestLogTestIntegration.java [hunk 2]
```diff
@@ -209,7 +235,17 @@ private String getRequestLogFilename() {
         return String.format("%s/%s", requestLogDirectory, "request.log");
     }
 
-    private String fileToString(final String filename) throws IOException {
-        return new String(Files.readAllBytes(new File(filename).toPath()));
+    private File getRequestLog() {
+        return new File(getRequestLogFilename());
+    }
+
+    private void clearRequestLog() throws IOException {
```
Comment (eshryane) [inline whois-api/src/test/java/net/ripe/db/whois/api/log/JettyRequestLogTestIntegration.java:242]: You can also use ```new FileWriter(getRequestLog()).close()``` to truncate the file.

File: whois-api/src/test/java/net/ripe/db/whois/api/log/JettyRequestLogTestIntegration.java (hunk 1 above)
Comment (petchells) [reply whois-api/src/test/java/net/ripe/db/whois/api/log/JettyRequestLogTestIntegration.java:112]: done in 2bc549bab

File: whois-api/src/test/java/net/ripe/db/whois/api/log/JettyRequestLogTestIntegration.java (hunk 2 above)
Comment (petchells) [reply whois-api/src/test/java/net/ripe/db/whois/api/log/JettyRequestLogTestIntegration.java:254]: done in 964696d65

File: whois-api/src/main/java/net/ripe/db/whois/api/httpserver/FilteredPasswordSlf4jRequestLogWriter.java [hunk 3]
```diff
@@ -1,17 +1,17 @@
...
 import org.eclipse.jetty.server.Slf4jRequestLogWriter;
 
 import javax.annotation.Nonnull;
 import java.io.IOException;
-import java.util.regex.Pattern;
 
-public class FilteredPasswordSlf4jRequestLogWriter extends Slf4jRequestLogWriter {
 
-    private static final Pattern PASSWORD_PATTERN = Pattern.compile("(?<=(?i)(password=))([^&]*)");
+public class FilteredPasswordSlf4jRequestLogWriter extends Slf4jRequestLogWriter {
```
Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/httpserver/FilteredPasswordSlf4jRequestLogWriter.java:10]: Rename back to `FilteredSlf4RequestLogWriter` as it's not just filtering passwords

File: whois-api/src/test/java/net/ripe/db/whois/api/log/JettyRequestLogTestIntegration.java [hunk 4]
```diff
@@ -152,11 +178,23 @@ public void password_filtered_case_insensitive() throws Exception {
...
 
+    @Test
+    public void override_filtered_case_insensitive() throws Exception {
+        RestTest.target(getPort(), "whois/test/person/TP1-TEST?oVeRrIdE=overrideUser,overPASS1")
+                .request()
+                .get(WhoisResources.class);
+
+
+        String actual = fileToString(getRequestLog());
+        assertThat(actual.toLowerCase(), containsString("override=overrideUser,FILTERED".toLowerCase()));
```
Comment (eshryane) [inline whois-api/src/test/java/net/ripe/db/whois/api/log/JettyRequestLogTestIntegration.java:194]: Should we test that the query parameter name has not changed (as in, match on `oVeRrIdE`). This allows us to detect if a user is not using all lowercase in a query param name.

File: whois-api/src/main/java/net/ripe/db/whois/api/httpserver/FilteredPasswordSlf4jRequestLogWriter.java (hunk 3 above)
Comment (petchells) [reply whois-api/src/main/java/net/ripe/db/whois/api/httpserver/FilteredPasswordSlf4jRequestLogWriter.java:10]: done in 44045027c

File: whois-api/src/test/java/net/ripe/db/whois/api/log/JettyRequestLogTestIntegration.java (hunk 4 above)
Comment (petchells) [reply whois-api/src/test/java/net/ripe/db/whois/api/log/JettyRequestLogTestIntegration.java:194]: yes, i think we should. i copied this test from the "PassWord" test but now you mention it, i don't think it's a good idea to 'lowerCase' everything in the test. updated in 44045027c

**Review by eshryane — APPROVED** [review]

**Pull request merged**